src/Experiment/src/Net/Http/LowLevel/HttpContentStream.cs

```diff
@@ -47,11 +49,15 @@ public class HttpContentStream : Stream, IEnhancedStream
         /// Instantiates a new <see cref="HttpContentStream"/>.
         /// </summary>
         /// <param name="request">The <see cref="ValueHttpRequest"/> to operate on.</param>
+        /// <param name="responseMessage">The <see cref="HttpResponseMessage"/> associated with this stream.</param>
+        /// <param name="headerSink">The <see cref="IHttpHeadersSink"/> for extract http headers.</param>
         /// <param name="ownsRequest">If true, the <paramref name="request"/> will be disposed once the <see cref="HttpContentStream"/> is disposed.</param>
-        public HttpContentStream(ValueHttpRequest request, bool ownsRequest)
+        public HttpContentStream(ValueHttpRequest request, HttpResponseMessage? responseMessage, IHttpHeadersSink? headerSink, bool ownsRequest)
```

Comment:
Instead of adding this to `HttpContentStream`, can we derive from it? This doesn't make much sense from a public API perspective and fits more as an implementation detail of the primitive handler.

Reply:
Resolved